ramips: add support for Netgear EX6100 v1

ysc3839 opened the pull request:
Specifications:
* SoC: MT7620A
* RAM: 64 MB DDR
* Flash: 8MB NOR SPI flash
* WiFi: MT7610E (5GHz) and builtin MT7620A (2.4GHz)
* LAN: 1x1000M (RTL8211E)

Installation:
The *-factory.chk images can be flashed from the
device's web interface or via nmrpflash.

Back to stock:
Use nmrpflash. Don't flash .chk firmware in OpenWrt.

This device is similar to EX6130, but uses RTL8211E for GbE LAN.

Thank @playya for his pull request #2286, this is based on it.
Thank @Icenowy for help making RTL8211E ethernet work.

5GHz WiFi is broken, maybe it's mt76 driver issue.
I don't have a device with GbE at this time so I don't know if GbE works. If you have this device please help test it, thanks!

ysc3839 commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,90 @@
+
+&ethernet {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &mdio_pins>;
```
I found there's `phy_reset_pins` in another device with RTL8211E:
https://github.com/openwrt/openwrt/blob/d2b8ccb1c04def81224da6f42f644c7d239b9986/target/linux/ramips/dts/mt7620a_edimax_ew-747x.dtsi#L123-L128 Now ethernet is working, is that necessary?
And it disabled phy0-4
https://github.com/openwrt/openwrt/blob/d2b8ccb1c04def81224da6f42f644c7d239b9986/target/linux/ramips/dts/mt7620a_edimax_ew-747x.dtsi#L139-L167 Should we need this too?

ysc3839 replied:
So anybody know if it's necessary?

adrianschmutzler replied:
Try to highlight the authors of the mentioned devices here ...

ysc3839 replied:
It seems that the original author doesn't have a GitHub account.
https://github.com/openwrt/openwrt/commit/88f7a29f99fb8209f6d7100d0cc0f7af6dba89cf

ysc3839 replied:
Ping @ynezz as he reviewed that patch.

adrianschmutzler replied:
Write an e-mail to Birger. He typically responds relatively fast and is willing to help if he can.

ysc3839 replied:
OK, sent.

ysc3839 replied:
@adrianschmutzler Still no reply from Birger Koblitz. Is there anyone else know this?

Neustradamus replied:
@ysc3839: Have you a reply from Birger Koblitz?
This GitHub account has been removed...

ysc3839 replied:
@Neustradamus No.
By the way thanks for your remind, I should close this PR since I have no access to this device for a long time.

ysc3839 commented:
There's another thing: there's a hardware gpio switch which controls AP and extender mode in stock firmware. Should I add it to dts?

ysc3839 commented:
After some test I just realized this devices does not support 802.11ac 80 MHz, set to 80 MHz causes clients failed to connect.
Is there any way to hide this option in luci? @jow-

ysc3839 commented:
Any updates?

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+	leds {
+		compatible = "gpio-leds";
+
+		led_power_green: power_g {
```
Use full name for colors:
led_power_green: power_green {

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+			gpios = <&gpio2 23 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_amber: power_a {
```
power_amber etc. below

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+		};
+
+		router_b {
+			label = "ex6100-v1:blue:router-arrow";
```
What's "router-arrow"?

ysc3839 replied:
![IMG_20200222_010100](https://user-images.githubusercontent.com/12028138/75054822-c74f7d80-550e-11ea-8525-e189ddaa4532.jpg)
It's these two "arrow shape" led.

ysc3839 replied:
![IMG_20200222_015619](https://user-images.githubusercontent.com/12028138/75058766-91ae9280-5516-11ea-8e46-174f10b781f5.jpg)
In user manual it's called Router/Client Arrow.
https://www.netgear.com/support/product/EX6100.aspx#docs

adrianschmutzler replied:
Okay, and what does it do?

ysc3839 replied:
I never seen these LEDs light before. And I can't find any description in user manual. I guess it indicates you should move closer to Router/Client for better signal?

adrianschmutzler replied:
Figure 1 tells you the names of the LEDs (if I got it right):

```
		router_link_green {
			label = "ex6100-v1:green:router-link";
			gpios = <&gpio2 25 GPIO_ACTIVE_LOW>;
		};

		router_link_red {
			label = "ex6100-v1:red:router-link";
			gpios = <&gpio2 24 GPIO_ACTIVE_LOW>;
		};

		router_arrow {
			label = "ex6100-v1:blue:router-arrow";
			gpios = <&gpio2 29 GPIO_ACTIVE_LOW>;
		};

		client_link_green {
			label = "ex6100-v1:green:client-link";
			gpios = <&gpio2 20 GPIO_ACTIVE_LOW>;
		};

		client_link_red {
			label = "ex6100-v1:red:client-link";
			gpios = <&gpio2 21 GPIO_ACTIVE_LOW>;
		};

		client_arrow {
			label = "ex6100-v1:blue:client-arrow";
			gpios = <&gpio2 22 GPIO_ACTIVE_LOW>;
		};
```

Despite, is it really "red" or "amber"?

adrianschmutzler replied:
> Despite, is it really "red" or "amber"?

I've just read the docs. So, I assume there are green and red LEDs and amber is when both of them are on?

adrianschmutzler replied:
> I never seen these LEDs light before. And I can't find any description in user manual. I guess it indicates you should move closer to Router/Client for better signal?

Okay, I also have found nothing. And if you do not have any indications based on what the device does with it runs with vendor OS, I'd just leave them unassigned so the user can do what he/she likes.

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+		};
+
+		device_b {
+			label = "ex6100-v1:blue:device-arrow";
```
What's device-arrow?

ysc3839 replied:
See picture above.

dengqf6 commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+		};
+	};
+
+	keys {
```
Missing "compatible" property.
It should be `compatible = "gpio-keys";`

ysc3839 replied:
It's defined in mt7620a_netgear_ex3700_ex6130.dtsi
https://github.com/openwrt/openwrt/blob/2f1cc5c3d5e35d6aa76e794e3d5b4f5856cd38bc/target/linux/ramips/dts/mt7620a_netgear_ex3700_ex6130.dtsi#L11-L12

adrianschmutzler replied:
Ah, so if you append to existing node, please do the following:

Change "keys" to "keys: keys" in DTSI. In DTS, just use the reference &keys then (out of root node) as you do it for &ethernet.

dengqf6 commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+	};
+
+	keys {
+		mode {
```
No reset button?

ysc3839 replied:
Defined in mt7620a_netgear_ex3700_ex6130.dtsi 
https://github.com/openwrt/openwrt/blob/2f1cc5c3d5e35d6aa76e794e3d5b4f5856cd38bc/target/linux/ramips/dts/mt7620a_netgear_ex3700_ex6130.dtsi#L14-L18

dengqf6 commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -154,6 +154,7 @@ netgear,wn3000rp-v3)
 	set_wifi_led "$boardname:green:router"
 	;;
 netgear,ex3700|\
+netgear,ex6100-v1|\
```
dts trigger for wireless can be used instead

ysc3839 replied:
Actually it's not WiFi LED, it indicates the connection to router (WAN) and clients (LAN). I just see EX6130 is here so I add this device too.

adrianschmutzler replied:
```
	ucidef_set_led_netdev "wlan5g" "ROUTER (green)" "$boardname:green:router" "wlan0"
	ucidef_set_led_netdev "wlan2g" "DEVICE (green)" "$boardname:green:device" "wlan1"
```

These are WiFi LEDs.

> I just see EX6130 is here so I add this device too.

So you are testifying that you just copy/paste from EX6130 no matter whether it is required or makes any sense, because you just don't check what you copy at all?

ysc3839 replied:
These are also not WiFi LEDs on EX3700 and EX6130. Both are link status LEDs.
![image](https://user-images.githubusercontent.com/12028138/75083270-e7089500-5552-11ea-8ff8-8461d1927ae1.png)
![image](https://user-images.githubusercontent.com/12028138/75083285-f38ced80-5552-11ea-8552-987da0f2537c.png)
I just think it's accepted on two devices, so I add this device too.

dengqf6 replied:
https://openwrt.org/docs/guide-user/hardware/led/generic.rssileds
You can make use of rssileds to show signal strength

ysc3839 commented:
> After some test I just realized this devices does not support 802.11ac 80 MHz, set to 80 MHz causes clients failed to connect.

I restored stock firmware and found it works with 80MHz. So this is an issue with mt76 driver? Did I missed something?

reversebias commented:
When was your fork? Looks like 19.07.1 has some improvements for ramips:
https://openwrt.org/releases/19.07/changelog-19.07.1
https://forum.openwrt.org/t/asus-rt-ac51u-support-for-5ghz/35041/18

I have one of these devices that I've been wanting to revive. Any tests I could help with?

ysc3839 commented:
@reversebias 
> This branch is 1 commit ahead, 87 commits behind openwrt:master.

I have not tested with 19.07, I'm using latest snapshot. I will try running on 19.07 to see if WiFi works.

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,98 @@
+/dts-v1/;
```
Please add a SPDX license identifier.

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -629,6 +629,21 @@ define Device/netgear_ex3700
+  BLOCKSIZE := 4k
+  IMAGE_SIZE := 7744k
+  IMAGES += factory.chk
+  IMAGE/factory.chk := $$(sysupgrade_bin) | check-size $$$$(IMAGE_SIZE) | \
```
check-size can be used without argument now. This will allow you to remove the line wrap as well.

adrianschmutzler commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -158,6 +158,14 @@ netgear,ex6130)
+	ucidef_set_led_rssi "rssiroutergreen" "RSSI ROUTER GREEN" "$boardname:green:router" "wwan" "34" "100"
+	ucidef_set_rssimon "wlan0" "200000" "1"
+	ucidef_set_led_rssi "rssiclientred" "RSSI CLIENT RED" "$boardname:red:client" "wlan0" "1" "66"
+	ucidef_set_led_rssi "rssiclientgreen" "RSSI CLIENT GREEN" "$boardname:green:client" "wlan0" "34" "100"
```
I don't think there is "wwan" in OpenWrt by default?
Congrats for how you reproduce the LED behavior though.
I'd keep the client part here (as it will out of the box). Just make sure you adjust the names if you follow my guidelines from above, e.g.
```
ucidef_set_led_rssi "clientlinkred" "Client Poor Link" "$boardname:red:client-link" "wlan0" "1" "66"
ucidef_set_led_rssi "clientlinkgreen" "Client Best Link" "$boardname:green:client-link" "wlan0" "34" "100"
```

Note that my names are meant as suggestions (based on the manual), I'm open for other ideas.

ysc3839 replied:
Because that LED is designed to indicate the connection with upstream router.

ysc3839 replied:
![image](https://user-images.githubusercontent.com/12028138/78578176-b611c480-7861-11ea-9900-85c092ce1b18.png)
http://www.downloads.netgear.com/files/GDC/EX6100/EX6100_UM_27Oct2015.pdf

ysc3839 replied:
Can `rssileds` update LEDs based on two wireless interface?

adrianschmutzler replied:
I understood what the LED is meant to do, however, in default OpenWrt configuration there is no "wwan", and thus this config won't make sense. In addition, rssileds package does not like being configured for an interface that doesn't exists and will cause issues with high load etc.

> Can rssileds update LEDs based on two wireless interface?

I don't know, I've never tried that. However, in any case this will only work for two _existing_ interfaces, so it still is no option for default config (unless wwan is present in buildbot snapshots and I'm just not aware).

ysc3839 replied:
Changed `wwan` to `wlan1`.

adrianschmutzler replied:
First: is it working? And do you still have "normal" load? Just curious...
Second: No. These LEDs are meant for connection to the AP, and not as a second client link LED. Despite, you told me that 5 GHz is not working anyway. So, since you have only one working Wiki anyway, just use that for the client link and fine.

adrianschmutzler commented:
If possible, it would be nice if you could provide more detailed instructions on flashing/recovery with nmrpflash, so that somebody knows what to do without using an external reference (at least in principle).

Despite, your commit message lacks a Signed-off-by.

ysc3839 commented:
> it would be nice if you could provide more detailed instructions on flashing/recovery with nmrpflash

I will write in the wiki after this merged.

adrianschmutzler commented:
> > it would be nice if you could provide more detailed instructions on flashing/recovery with nmrpflash
> 
> I will write in the wiki after this merged.

I'd also like to have it in the commit message (at least in a very abbreviated form). This way it is permanent, while the Wiki might change or become unavailable.

masterblaster79 commented:
Hi.
I am very interested in this and tried to build it.
The sysupgrade.bin image was not created in my case. "Image too big" or similar.

I did the following changes (added IMAGE_SIZE and removed the check-size parameter) to mt7620.mk to make it work:
```
define Device/netgear_ex6100-v1
  SOC := mt7620a
  NETGEAR_BOARD_ID := U12H248T00_NETGEAR
  BLOCKSIZE := 4k
  IMAGES += factory.chk
  IMAGE/factory.chk := $$(sysupgrade_bin) | check-size 7744k | \
	netgear-chk
  DEVICE_PACKAGES := kmod-mt76x0e rssileds
  DEVICE_VENDOR := NETGEAR
  DEVICE_MODEL := EX6100
  DEVICE_VARIANT := v1
endef
TARGET_DEVICES += netgear_ex6100-v1
```

to:
```
define Device/netgear_ex6100-v1
  SOC := mt7620a
  NETGEAR_BOARD_ID := U12H248T00_NETGEAR
  BLOCKSIZE := 4k
  IMAGE_SIZE := 7744k
  IMAGES += factory.chk
  IMAGE/factory.chk := $$(sysupgrade_bin) | check-size | netgear-chk
  DEVICE_PACKAGES := kmod-mt76x0e rssileds
  DEVICE_VENDOR := NETGEAR
  DEVICE_MODEL := EX6100
  DEVICE_VARIANT := v1
endef
TARGET_DEVICES += netgear_ex6100-v1
```

I did not flash it yet though. I'll report back once done.

masterblaster79 commented:
It's running now, 5GHZ wifi doesn't work. For now, no issues.

ysc3839 commented:
@atiensivu How do you know that "power booster"? From stock firmware source code? Can you provide more information?

atiensivu commented:
> @atiensivu How do you know that "power booster"? From stock firmware source code? Can you provide more information?

I will have to find the documentation but I believe it was posted either on WikiDev or the OpenWRT page. I will try to find it again. I believe someone tore one apart and found it. The radios don't seem very powerful with the default firmware so I'm not 100% sure that is accurate (a power booster existing).

atiensivu commented:
> > @atiensivu How do you know that "power booster"? From stock firmware source code? Can you provide more information?
> 
> I will have to find the documentation but I believe it was posted either on WikiDev or the OpenWRT page. I will try to find it again. I believe someone tore one apart and found it. The radios don't seem very powerful with the default firmware so I'm not 100% sure that is accurate (a power booster existing).

"5GHz Power Amplifier Module;Skyworks;SE5003L1;;1;"

https://wikidevi.wi-cat.ru/Netgear_EX6100

I doubt Netgear posted any source for their changes but I can always look.

masterblaster79 commented:
Not sure if this helps, and I didn't look into this, but Netgear has published source code for the device here:
[https://kb.netgear.com/2649/NETGEAR-Open-Source-Code-for-Programmers-GPL](https://kb.netgear.com/2649/NETGEAR-Open-Source-Code-for-Programmers-GPL)

ysc3839 commented:
@masterblaster79 The WiFi driver is close-sourced.

masterblaster79 commented:
I now have an uptime of 75 days without any issue using this device as simple AP only. It's running r13372-afea16b8f7.

Other than the missing 5G wifi, the following two things I noticed:

1.
root@APOGNEW:~# dmesg
dmesg: klogctl: Function not implemented

2.
network -> switch:
Switch switch0 has an unknown topology - the VLAN settings might not be accurate.
..and it shows five switch ports

dangowrt commented:
Can you try building my [staging tree](https://git.openwrt.org/?p=openwrt/staging/dangole.git;a=commit;h=d5cc097c22fb82eb87211e1d82d6bb10126d0b2c) with this device and report back on how wifi performance changes?

ysc3839 commented:
@dangowrt I will have a try.

masterblaster79 commented:
@dangowrt 
I flashed your tree.
2.4 GHZ works as before (haven't done extensive tests though) but 5 GHZ interface is completely gone now (before it was there but not working).
As before I cannot see the kernel messages.
root@APOG:~# dmesg
dmesg: klogctl: Function not implemented

masterblaster79 commented:
$ grep -i printk build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/linux-5.4.68/.config
# CONFIG_PRINTK is not set

..is this the reason for not seeing any kernel messages?

masterblaster79 commented:
Hi again.

Good news:
Other than stated before, the 5GHZ wifi is working now using a build from the dangowrt tree.

First I thought I lost the 5GHZ interface, but after I reset to factory defaults and disconnected/reconnected the device from power the 5GHZ interface reappeared and works. Nice.

Setting CONFIG_KERNEL_PRINTK gave me the kernel messages back btw..

I am happy for now.

ysc3839 commented:
I just resolved conflicts, please review.

ysc3839 commented:
@dangowrt I tested your staging tree, the WiFi is still broken. I have to try many time to connect to 5GHz AP. And both 2.4GHz and 5GHz are slow. The TX bandwith is only 27.0 Mbits/sec tested with iperf3.

adschm commented on target/linux/ramips/dts/mt7620a_netgear_ex6100-v1.dts:
```diff
@@ -0,0 +1,57 @@
+
+&leds {
+	router_arrow {
+		label = "ex6100-v1:blue:router-arrow";
```
Please drop the model prefix, use just "blue:router-arrow", same below.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -158,6 +158,14 @@ netgear,ex6130)
+	ucidef_set_led_rssi "routerlinkgreen" "Router Best Link" "$boardname:green:router" "wlan1" "34" "100"
+	ucidef_set_rssimon "wlan0" "200000" "1"
+	ucidef_set_led_rssi "clientlinkred" "Client Poor Link" "$boardname:red:device" "wlan0" "1" "66"
+	ucidef_set_led_rssi "clientlinkgreen" "Client Best Link" "$boardname:green:device" "wlan0" "34" "100"
```
drop $boardname: here for all lines.

ysc3839 replied:
Fixed.

dangowrt commented:
> @dangowrt I tested your staging tree, the WiFi is still broken. I have to try many time to connect to 5GHz AP. And both 2.4GHz and 5GHz are slow. The TX bandwith is only 27.0 Mbits/sec tested with iperf3.

My changes only touched the 2.4GHz part (which is built-in in MT7620 SoC and controlled by the rt2x00 driver in OpenWrt), 5GHz is using an additional chip (MT7610E) which is supported by the mt76 driver (but known not to work very well).
Hence it'd be interesting to me how my staging tree performs in comparison to recent unmodified OpenWrt in the 2.4GHz band. 27.0MBit/sec (single or multi-stream, btw?) is not great, but already much better than the ~6-8MBit/s users were reporting to see with MT7620A recently (but that can be related to board design and many other factors as well).

ysc3839 commented:
@dangowrt Sorry, I forget to say 27.0 Mbits/sec is tested in 5GHz. I have not tested with unmodified OpenWrt yet.

masterblaster79 commented:
I did some measurements using iperf installed on the device and yes, the performance is not good. (Client Intel AC 8265)

dangowrt tree, 2.4G: ~48Mbit Tx, 30-35Mbit Rx
openwrt tree,   2.4G: ~50Mbit Tx, 25-35Mbit Rx
Almost the same.

After flashing the official openwrt, I again lost the 5G interface. Even going back to dangowrt didn't bring it back. 5G is gone for now. 5G performance anyway was even worse than 2.4G ~30-40Mbit Tx and ~20Mbit Rx

Ethernet only negotiates 100Mbit with the switch - shouldn't this be 1000Mbit?

ysc3839 committed:
ramips: add support for Netgear EX6100 v1

Specifications:
* SoC: MT7620A
* RAM: 64 MB DDR
* Flash: 8MB NOR SPI flash
* WiFi: MT7610E (5GHz) and builtin MT7620A (2.4GHz)
* LAN: 1x1000M (RTL8211E)

5GHz WiFi is broken, maybe it's mt76 driver issue.
LAN GbE is not tested.

Installation:
The *-factory.chk images can be flashed from the
device's web interface or via nmrpflash.

Back to stock:
Use nmrpflash. Don't flash .chk firmware in OpenWrt.

This device is similar to EX6130, but uses RTL8211E for GbE LAN.

Signed-off-by: Richard Yu <yurichard3839@gmail.com>

ysc3839 commented:
@masterblaster79 
> I again lost the 5G interface

Maybe you selected wrong WiFi driver. EX6100 has a MT7610E WiFi chip, which requires `kmod-mt76x0e` package.

> Ethernet only negotiates 100Mbit with the switch - shouldn't this be 1000Mbit?

I don't have a GbE adapter here, so I don't know if it works.

ysc3839 commented:
@dangowrt Some test result, this is tested with unmodified OpenWrt:
Encryption: WPA2 PSK (CCMP)
Client: Realtek 8822CE with Windows 10 2004
iperf3:
```
2.4GHz
$ iperf3 -c 192.168.1.1
6.07 Mbits/sec
$ iperf3 -c 192.168.1.1 -R
3.07 Mbits/sec

5GHz
$ iperf3 -c 192.168.1.1
100 Mbits/sec
$ iperf3 -c 192.168.1.1 -R
114 Mbits/sec
```

I tried a lot of times connecting to 5GHz, here is the system log when connecting: (MAC address was removed)
```
root@OpenWrt:~# logread -f
Tue Oct  6 08:00:15 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:16 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:17 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:18 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:21 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:22 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:23 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:00:24 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:00 2020 cron.err crond[1583]: USER root pid 2525 cmd /usr/sbin/ffwatchd running || /etc/init.d/freifunk-watchdog restart
Tue Oct  6 08:01:10 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:10 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:10 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:10 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:11 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:11 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:11 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:11 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:12 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:22 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: deauthenticated due to local deauth request
Tue Oct  6 08:01:49 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:50 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:51 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:52 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:55 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:55 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:01:55 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:55 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:55 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:55 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:56 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:56 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:01:57 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:01:59 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 1
Tue Oct  6 08:02:00 2020 cron.err crond[1583]: USER root pid 2791 cmd /usr/sbin/ffwatchd running || /etc/init.d/freifunk-watchdog restart
Tue Oct  6 08:02:00 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:02:01 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:02:02 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:02:02 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: authenticated
Tue Oct  6 08:02:03 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:02:03 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:02:03 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:02:03 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:02:03 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
Tue Oct  6 08:02:03 2020 daemon.notice hostapd: wlan0: STA-OPMODE-N_SS-CHANGED 8c:**:**:**:**:** 2
Tue Oct  6 08:02:04 2020 daemon.info hostapd: wlan0: STA 8c:**:**:**:**:** IEEE 802.11: associated (aid 1)
```
The client shows in luci, but OS said can't connect:
![image](https://user-images.githubusercontent.com/12028138/95212491-5603d700-0820-11eb-85f0-a68a4177c6f0.png)
![image](https://user-images.githubusercontent.com/12028138/95212576-6d42c480-0820-11eb-83c6-4df17af8087e.png)

ysc3839 commented:
@dangowrt With your staging tree:
```
2.4GHz
$ iperf3 -c 192.168.1.1
4.19 Mbits/sec
$ iperf3 -c 192.168.1.1 -R
2.70 Mbits/sec

5GHz
$ iperf3 -c 192.168.1.1
143 Mbits/sec
$ iperf3 -c 192.168.1.1 -R
117 Mbits/sec
```

5GHz has same issue as unmodified OpenWrt.

masterblaster79 commented:
> > I again lost the 5G interface
> 
> Maybe you selected wrong WiFi driver. EX6100 has a MT7610E WiFi chip, which requires `kmod-mt76x0e` package.
> 
Well, no. I flashed the same .bin which had 5G before.
Also the interface is there and configurable, but it just doesn't work.
```
root@apog:~# opkg list-installed | grep kmod-mt
kmod-mt76-core - 5.4.68+2020-09-23-b22977c2-2
kmod-mt76x0-common - 5.4.68+2020-09-23-b22977c2-2
kmod-mt76x02-common - 5.4.68+2020-09-23-b22977c2-2
kmod-mt76x0e - 5.4.68+2020-09-23-b22977c2-2
root@apog:~# ls -l /lib/firmware/mediatek/
-rw-r--r--    1 root     root         80680 Sep 30 19:57 mt7610e.bin
```
> $ iperf3 -c 192.168.1.1 -R
> 2.70 Mbits/sec

Ouch, my 2.4GHZ performance is much better than yours as it seems.

ysc3839 commented:
@masterblaster79 
> but it just doesn't work

What's wrong with it?

masterblaster79 commented:
> What's wrong with it?

Thy symptom is that the SSID is not broadcasted and clients cannot connect. I have no spectrum analyzer or something like that, but it looks like it just doesn't send anything.

ysc3839 commented:
@masterblaster79 I think I have same issue as yours.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -158,6 +158,14 @@ netgear,ex6130)
+	ucidef_set_rssimon "wlan0" "200000" "1"
+	ucidef_set_led_rssi "clientlinkred" "Client Poor Link" "red:device" "wlan0" "1" "66"
+	ucidef_set_led_rssi "clientlinkgreen" "Client Best Link" "green:device" "wlan0" "34" "100"
+	;;
```
This is different from the intended use of the LEDs as it seems. I'd prefer to leave these unset, so the user can choose what to do.

adschm commented:
What's the status of this device? Is it still RFT? Is it ready for review?

ysc3839 commented:
@adschm I have no access to this device so I don't know if Wi-Fi is still broken. It's ready for review since I can't do much about the Wi-Fi driver.

Neustradamus commented:
@ysc3839: Why have you closed it?

Neustradamus commented:
It is better to have a better solution that close this PR :/

ysc3839 commented:
@Neustradamus Because I have no access to this device for a long time. I can't test newer version on this device. And this PR is inactive for a long time.
